x-pack/plugins/observability/public/components/shared/exploratory_view/series_editor/columns/breakdowns.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+import { EuiSuperSelect } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FieldLabels } from '../../configurations/constants';
+import { useUrlStorage } from '../../hooks/use_url_strorage';
+
+interface Props {
+  seriesId: string;
+  breakdowns: string[];
+}
+
+export function Breakdowns({ seriesId, breakdowns = [] }: Props) {
+  const items = breakdowns.map((breakdown) => ({ id: breakdown, label: FieldLabels[breakdown] }));
+
+  const { setSeries, series } = useUrlStorage(seriesId);
+
+  const selectedBreakdown = series.breakdown;
+  const NO_BREAKDOWN = 'no_breakdown';
+
+  const onOptionChange = (optionId: string) => {
+    if (optionId === NO_BREAKDOWN) {
+      setSeries(seriesId, {
+        ...series,
+        breakdown: undefined,
+      });
+    } else {
+      setSeries(seriesId, {
+        ...series,
+        breakdown: selectedBreakdown === optionId ? undefined : optionId,
+      });
+    }
+  };
+
+  useEffect(() => {}, [selectedBreakdown]);
+
+  items.push({
```

Comment:
Is this meant to be in the main component body? Won't this cause another item to be pushed onto the array on each rerender. Is that intentional?

Reply:
i reordered the code to more clear